packages/odyssey-react/src/components/Select/Select/index.tsx

```diff
@@ -0,0 +1,178 @@
+   */
+  children: ReactElement | ReactElement[];
+
+  /**
+   * The form field hint
+   */
+  hint?: string,
+
+  /**
+   * The form field label
+   */
+  label: string,
+
+  /**
+   * The underlying select element id attribute. Automatically generated if not provided
+   */
+  id?: string,
+
+  /**
+   * Text to display when the select is optional, i.e. required prop is false
+   */
+  optionalLabel?: string,
+
+  /**
+   * The underlying select element name attribute for the group
+   */
+  name: string,
+
+  /**
+   * The underlying select element required attribute for the group
+   * @default true
+   */
+  required?: boolean,
+
+  /**
+   * The underlying select element disabled attribute for the group
+   * @default false
+   */
+  disabled?: boolean,
+
+  /**
+   * The selected option value attribute for a controlled group.
+   */
+  value?: string,
+
+  /**
+   * Callback executed when the select fires a change event
+   * @param {Object} event the event object
+   * @param {string} value the string value of the select
+   */
+  onChange?: (event?: ChangeEvent<HTMLSelectElement>, value?: string) => void,
+}
+
+/**
+ * Often referred to as a "dropdown menu" this input triggers a menu of
+ * options a user can select.
+ */
+const Select = forwardRefWithStatics<
+  HTMLSelectElement,
+  Props,
+  Statics
+>((props, ref) => {
+  const {
+    id,
+    hint,
+    children,
+    disabled = false,
+    label,
+    optionalLabel,
+    name,
+    onChange,
+    required = true,
+    value,
+    ...rest
+  } = props;
+
+  const omitProps = useOmit(rest);
+
+  const oid = useOid(id);
+
+  useChoices(oid, value);
+
+  const isOptional = (!required && optionalLabel) ? true : null;
+
+  const labelElement = (
+    <label
+      className="ods-label"
+      htmlFor={oid}
+    >
+      {label}
+      { isOptional && <span
+        className="ods-label--optional"
+        children={optionalLabel}
+      />}
+    </label>
+  );
+
+  const handleChange = useCallback((
+    event: ChangeEvent<HTMLSelectElement>
+  ) => {
+    onChange?.(event, event.target.value);
+  },
+    [onChange]
+  );
+
+
+  const selectElement = (
+    // eslint-disable-next-line jsx-a11y/no-onchange
+    <select
+      {...omitProps}
+      id={oid}
+      name={name}
+      disabled={disabled}
+      required={required}
+      onChange={handleChange}
+      value={value}
+      ref={ref}
+    >
+      {children}
+    </select>
+  );
+
+  const hintElement = hint && (
+    <aside
+      className="ods-field--hint"
+      children={hint}
+    />
+  );
+
+  return (
+    <fieldset data-optional={isOptional} className="ods-fieldset">
+      <div className="ods-fieldset-flex">
+        {selectElement}
+        {labelElement}
+        {hintElement}
+      </div>
+    </fieldset>
+  );
+});
+
+interface Statics {
+  Option: typeof SelectOption;
+  OptionGroup: typeof SelectOptionGroup;
+}
+
+Select.Option = SelectOption;
+Select.OptionGroup = SelectOptionGroup;
```

Comment:
I like this. Some of the other component files, like Modal, are too long! I think it's worth updating to follow what you've done here.